serde-support/src/main/java/io/micronaut/serde/support/deserializers/ObjectDeserializer.java

```diff
@@ -155,7 +155,12 @@ public <T> DeserBean<T> getDeserializableBean(Argument<T> type, DecoderContext d
         // Use suppliers to prevent recursive update because the lambda can call the same method again
         Supplier<DeserBean<?>> deserBeanSupplier = deserBeanMap.computeIfAbsent(key, ignore -> SupplierUtil.memoizedNonEmpty(() -> createDeserBean(type, serdeArgumentConf, decoderContext)));
         DeserBean<?> deserBean = deserBeanSupplier.get();
-        deserBean.initialize(decoderContext);
+        // Double check locking
+        if (!deserBean.initialized) {
+            synchronized (this) {
```

Comment:
maybe switch to reentrantlock

Reply:
yes

Reply:
Pls check now

Reply:
nice, also makes it more clear that initialize needs a lock.